### PR TITLE
Implement Blik component

### PR DESCRIPTION
**Pull request by jreij**
## Summary
- Create a Blik component and add it to the Drop in
- Allow AwaitView to support different message texts, per payment method

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 1]
```diff
@@ -186,6 +192,7 @@ internal fun getProviderForType(type: String): PaymentComponentProvider<PaymentC
         PaymentMethodTypes.SCHEME -> CardComponent.PROVIDER
         PaymentMethodTypes.SEPA -> SepaComponent.PROVIDER
         PaymentMethodTypes.WECHAT_PAY_SDK -> WeChatPayComponent.PROVIDER
+        PaymentMethodTypes.BLIK -> BlikComponent.PROVIDER
```
Comment (caiofaustino) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:195]: A bit nit picky, but maybe add this in alphabetic order, the same order the modules show up on the IDE, to make this ugly code a bit easier to read at least 😅

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 2]
```diff
@@ -295,6 +302,10 @@ internal fun getComponentFor(
             val weChatPayConfiguration: WeChatPayConfiguration = dropInConfiguration.getConfigurationFor(PaymentMethodTypes.WECHAT_PAY_SDK, context)
             WeChatPayComponent.PROVIDER.get(fragment, paymentMethod, weChatPayConfiguration)
         }
+        PaymentMethodTypes.BLIK -> {
```
Comment (caiofaustino) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:305]: same as above

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 3]
```diff
@@ -331,6 +342,7 @@ internal fun getViewFor(
         PaymentMethodTypes.OPEN_BANKING -> OpenBankingRecyclerView(context)
         PaymentMethodTypes.SCHEME -> CardView(context)
         PaymentMethodTypes.SEPA -> SepaView(context)
+        PaymentMethodTypes.BLIK -> BlikView(context)
```
Comment (caiofaustino) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:345]: same as above

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt [hunk 4]
```diff
@@ -288,26 +289,37 @@ class DropInConfiguration : Configuration, Parcelable {
         }
 
         /**
-         * Add configuration for Sepa payment method.
+         * Add configuration for BCMC payment method.
```
Comment (caiofaustino) [inline drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt:292]: Thanks for catching these 😄

File: blik-base/src/main/java/com/adyen/checkout/blik/BlikOutputData.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @NonNull
+    public ValidatedField<String> getBlikCodeField() {
+        return mBlikCodeField;
+    }
+
+    private ValidatedField.Validation getBlikCodeValidation(@NonNull String blikCode) {
+        if (TextUtils.isEmpty(blikCode)) {
+            return ValidatedField.Validation.PARTIAL;
```
Comment (caiofaustino) [inline blik-base/src/main/java/com/adyen/checkout/blik/BlikOutputData.java:43]: You were asking me why empty field was not showing validation error on the UI, this is probably why. 
Just a UX choice, you can choose to show the error message for partial validation on the View.

**Review by caiofaustino — APPROVED** [review]

File: blik-base/src/main/java/com/adyen/checkout/blik/BlikOutputData.java (hunk 5 above)
Comment (jreij) [reply blik-base/src/main/java/com/adyen/checkout/blik/BlikOutputData.java:43]: My question was mainly about after you press the Pay button, but then I found out that the issue only happens in v3 and it seems to be fixed in v4 :+1:

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt (hunk 1 above)
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:195]: Thanks, I didn't notice they were sorted 😅

**Review by caiofaustino — APPROVED** [review]

**Pull request merged**